src/features/processPicker.ts

```diff
@@ -75,6 +84,122 @@ export class RemoteAttachPicker {
+        let pipeProgram: string = pipeTransport.pipeProgram;
+        let pipeArgs: string[] | string = pipeTransport.pipeArgs;
+        let quoteArgs: boolean = pipeTransport.quoteArgs != null ? pipeTransport.quoteArgs : true; // default value is true
+        let platformSpecificPipeTransportOptions: IPipeTransportOptions = this.getPlatformSpecificPipeTransportOptions(pipeTransport, osPlatform); 
+
+        if (platformSpecificPipeTransportOptions) {
+            pipeProgram = platformSpecificPipeTransportOptions.pipeProgram || pipeProgram;
+            pipeArgs = platformSpecificPipeTransportOptions.pipeArgs || pipeArgs;
+            quoteArgs = platformSpecificPipeTransportOptions.quoteArgs != null ? platformSpecificPipeTransportOptions.quoteArgs : quoteArgs;
+        }
+
+        return {
+            pipeProgram: pipeProgram,
+            pipeArgs: pipeArgs,
+            quoteArgs: quoteArgs
+        };
+    }
+
+    // If the current process is on a current operating system and a specific pipe transport
+    // is included, then use that specific pipe transport configuration.
+    //
+    // Note: osPlatform is passed as an argument for testing.
+    private static getPlatformSpecificPipeTransportOptions(config: any, osPlatform: string): IPipeTransportOptions {
+        if (osPlatform == "darwin" && config.osx) {
+            return config.osx;
+        } else if (osPlatform == "linux" && config.linux) {
+            return config.linux;
+        } else if (osPlatform == "win32" && config.windows) {
+            return config.windows;
+        }
+
+        return null;
+    }
+
+    // Creates a pipe command string based on the type of pipe args.
+    private static createPipeCmd(pipeProgram: string, pipeArgs: string | string[], quoteArgs: boolean): Promise<string> {
+        return this.ValidateAndFixPipeProgram(pipeProgram).then(fixedPipeProgram => {
+            if (typeof pipeArgs == "string") {
+                return Promise.resolve(this.createPipeCmdFromString(fixedPipeProgram, pipeArgs, quoteArgs));
+            }
+            else if (pipeArgs instanceof Array) {
+                return Promise.resolve(this.createPipeCmdFromArray(fixedPipeProgram, pipeArgs, quoteArgs));
+            } else {
+                // Invalid args type
+                return Promise.reject<string>(new Error("pipeArgs must be a string or a string array type"));
+            }
+        });
+    }
+
+    public static createPipeCmdFromString(pipeProgram: string, pipeArgs: string, quoteArgs: boolean): string {
+        // Quote program if quoteArgs is true.
+        let pipeCmd: string = this.quoteArg(pipeProgram);
+
+        // If ${debuggerCommand} exists in pipeArgs, replace. No quoting is applied to the command here.
+        if (pipeArgs.indexOf(this.debuggerCommand) >= 0) {
+            pipeCmd = pipeCmd.concat(" ", pipeArgs.replace(/\$\{debuggerCommand\}/g, this.scriptShellCmd));
```

Comment:
I think we want to keep it case-sensitive if we want to be similar to commands in VSCode such as ${csharp:ProcessPicker}.